crates/core/tedge_actors/src/builders.rs

```diff
@@ -113,24 +97,20 @@ pub struct NoConfig;
 ///
 /// An actor whose builder is a `MessageSink<M, C>` can be connected to any other actor
 /// whose builder is a `MessageSource<M, C>` so that the sink can receive messages from that source.
-///
-/// A sink might be interested only in a subset of the messages emitted by the source.
-/// For that purpose each source implementation defines a `Config` type parameter,
-/// and the sink has to provide the configuration value specific to its needs.
-pub trait MessageSink<M: Message, Config> {
-    /// Return the config used by this actor to connect the message source
-    fn get_config(&self) -> Config;
-
+pub trait MessageSink<M: Message> {
     /// Return the sender that can be used by peers to send messages to this actor
     fn get_sender(&self) -> DynSender<M>;
 
     /// Add a source of messages to the actor under construction
-    fn add_input<N>(&mut self, source: &mut impl MessageSource<N, Config>)
+    ///
+    /// A sink might be interested only in a subset of the messages emitted by the source.
+    /// This subset is defined by the config parameter.
+    fn add_input<N, C>(&mut self, config: C, source: &mut impl MessageSource<N, C>)
```

Comment:
Something for a follow-up PR: I personally find this `add_input` helper that takes a config counter-intuitive. In my head, the config is something meant for the source which is rightly captured in `MessageSource::register_peer`. So, I'd rather have the users use the source's `register_peer` API rather than this confusing indirection.

And if we're keeping it, I'd rename it to `add_source` or `connect_source` to be in sync with its `add_sink` counterpart.

Reply:
I'm fully aligned. And as even been tempted to make this change in the same PR. I decided otherwise - postponing this change to a follow-up PR focused on naming + docs.